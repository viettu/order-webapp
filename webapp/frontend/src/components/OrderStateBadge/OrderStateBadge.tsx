import { Badge } from '@chakra-ui/react';
import { OrderStates } from '../../data';

type OrderStateBadgeProps = {
  fontSize?: string;
  state: string;
};

const OrderStateBadge: React.FC<OrderStateBadgeProps> = ({ fontSize, state }) => {
  return (
    <>
      {state === OrderStates.CREATED && <Badge fontSize={fontSize}>CREATED</Badge>}
      {state === OrderStates.CONFIRMED && (
        <Badge fontSize={fontSize} colorScheme="green">
          CONFIRMED
        </Badge>
      )}
      {state === OrderStates.CANCELLED && (
        <Badge fontSize={fontSize} colorScheme="red">
          CANCELLED
        </Badge>
      )}
      {state === OrderStates.DELIVERED && (
        <Badge fontSize={fontSize} colorScheme="purple">
          {' '}
          DELIVERE
        </Badge>
      )}
    </>
  );
};

OrderStateBadge.defaultProps = {
  fontSize: '1.2em',
};

export default OrderStateBadge;
