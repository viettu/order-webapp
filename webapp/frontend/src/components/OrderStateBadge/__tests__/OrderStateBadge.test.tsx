import React from 'react';
import { OrderStateBadge } from '../OrderStateBadge';
import renderer from 'react-test-renderer';
import { TestComponentWrapper } from '../../../utils';

test('snapshot - CANCELLED', () => {
  const component = renderer.create(
    <TestComponentWrapper>
      <OrderStateBadge state="CANCELLED"></OrderStateBadge>
    </TestComponentWrapper>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('snapshot - CREATED', () => {
  const component = renderer.create(
    <TestComponentWrapper>
      <OrderStateBadge state="CREATED"></OrderStateBadge>
    </TestComponentWrapper>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('snapshot - DELIVERED', () => {
  const component = renderer.create(
    <TestComponentWrapper>
      <OrderStateBadge state="DELIVERED"></OrderStateBadge>
    </TestComponentWrapper>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
