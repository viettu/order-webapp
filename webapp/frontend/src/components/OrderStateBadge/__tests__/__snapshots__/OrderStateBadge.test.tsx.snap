// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot - CANCELLED 1`] = `
Array [
  <span
    className="chakra-badge css-1j2zy3h"
  >
    CANCELLED
  </span>,
  <span />,
  <span
    className="chakra-env"
    hidden={true}
  />,
]
`;

exports[`snapshot - CREATED 1`] = `
Array [
  <span
    className="chakra-badge css-pst2fr"
  >
    CREATED
  </span>,
  <span />,
  <span
    className="chakra-env"
    hidden={true}
  />,
]
`;

exports[`snapshot - DELIVERED 1`] = `
Array [
  <span
    className="chakra-badge css-70togv"
  >
     
    DELIVERED
  </span>,
  <span />,
  <span
    className="chakra-env"
    hidden={true}
  />,
]
`;
